### PR TITLE
Dialog new archive improvements

### PR DESCRIPTION
**Pull request by rbuj**

### CODE REVIEW

File: src/dlg-new.c [hunk 1]
```diff
@@ -364,42 +358,18 @@ dlg_new_archive (FrWindow  *window,
...
-	g_signal_connect (G_OBJECT (data->dialog),
-			  "destroy",
-			  G_CALLBACK (destroy_cb),
-			  data);
+        gtk_builder_add_callback_symbols (data->builder,
+                                          "on_dialog_destroy", G_CALLBACK (destroy_cb),
+                                          "on_n_password_entry_changed", G_CALLBACK (password_entry_changed_cb),
+                                          "on_n_volume_checkbutton_toggled", G_CALLBACK (volume_toggled_cb),
+                                          NULL);
+        gtk_builder_connect_signals (data->builder, data);
```
Comment (sc0w) [inline src/dlg-new.c]: wrong indentation

Comment (rbuj) [reply src/dlg-new.c]: I do not agree. Are you sure that it has a bad indentation? I can't see it.
Can another team member review it?

Comment (rbuj) [reply src/dlg-new.c]: now I see it... you mean the line must start with a tab and then it must be aligned with what is inside the parentheses using spaces.

**Comment by rbuj** [comment]
@sc0w done

**Comment by sc0w** [comment]
yes, thanks, is not easy for me to explain wrong indentations, I try to explain me with screenshots :)

**Comment by rbuj** [comment]
The file header sets how many spaces should a tab correspond to. For that reason I saw it well aligned on the editor (nano). I apologize for the inconvenience. ;-)

**Comment by sc0w** [comment]
This PR is for this dialog?

I don't know why, but gtk inspector says me the dialog with this PR is still using stock icons.

**Comment by sc0w** [comment]
```
$ grep -iR "create-archive"
src/fr-stock.h:#define FR_STOCK_CREATE_ARCHIVE "create-archive"
```
```
$ grep -iR "FR_STOCK_CREATE_ARCHIVE"
src/dlg-batch-add.c:	gtk_button_set_label (GTK_BUTTON (GET_WIDGET ("a_add_button")), FR_STOCK_CREATE_ARCHIVE);
src/fr-stock.c:	{ FR_STOCK_CREATE_ARCHIVE, "add-files-to-archive" },
src/fr-stock.c:	{ FR_STOCK_CREATE_ARCHIVE, N_("C_reate"), 0, 0, GETTEXT_PACKAGE },
src/fr-stock.h:#define FR_STOCK_CREATE_ARCHIVE "create-archive"
```

probably the icons in the ui file are never used, it seems to handled by code

**Comment by rbuj** [comment]
> This PR is for this dialog?

No, it isn't. Launch engrampa, next click on New icon (the blank document icon with a star, which is located on the icon bar), or go to Archive -> New..., or press Ctrl+N.

**Comment by rbuj** [comment]
> I don't know why, but gtk inspector says me the dialog with this PR is still using stock icons.

I recall that this dialog uses the file batch-add-files.ui. It's on the to-do list ;-)

**Comment by rbuj** [comment]
To-Do list:

```
Task. Upgrade UI files from Gtk2 to Gtk3

To-Do mate-applets/stickynotes/stickynotes.ui:  <requires lib="gtk+" version="2.16"/>
To-Do mate-applets/cpufreq/cpufreq-preferences.ui:  <requires lib="gtk+" version="2.16"/>
To-Do mate-applets/trashapplet/trashapplet-empty-progress.ui:  <requires lib="gtk+" version="2.16"/>
To-Do mate-applets/battstat/battstat_applet.ui:  <requires lib="gtk+" version="2.16"/>
(50%: remove CPU wakeups) mate-power-manager/data/gpm-statistics.ui:  <requires lib="gtk+" version="2.16"/>
PR mate-power-manager/data/gpm-prefs.ui:  <requires lib="gtk+" version="2.16"/>
PR mate-utils/logview/data/logview-filter.ui:  <requires lib="gtk+" version="2.16"/>
...

Task. Engrampa - Remove Stock Icons: Custom / General Availability (GA)

To-Do engrampa/po/POTFILES.in:engrampa/src/fr-stock.h
To-Do engrampa/src/Makefile.am:        fr-stock.h                      \
To-Do engrampa/src/dlg-batch-add.c:#include "fr-stock.h"
To-Do engrampa/src/fr-init.c:#include "fr-stock.h"
To-Do engrampa/src/fr-stock.c:#include "fr-stock.h"
To-Do engrampa/src/fr-window.c:#include "fr-stock.h"
To-Do engrampa/src/ui.h:#include "fr-stock.h"
PR engrampa/src/dlg-add-files.c:#include "fr-stock.h"
PR engrampa/src/dlg-add-folder.c:#include "fr-stock.h"
PR engrampa/src/dlg-extract.c:#include "fr-stock.h"
...
```

**Comment by sc0w** [comment]
I noticed one problem, not related to this PR, I see the same behavior in engrampa 1.22.3

how to reproduce: press tab several times, and the focus stops here:

**Comment by rbuj** [comment]
using the TAB key to autocomplete. type any key next press tab key...

**Comment by sc0w** [comment]
ok, I think this is an accesibility issue, I expected to change between widgets, <kbd>control</kbd><kbd>shift</kbd><kbd>tab</kbd> works as expected

for example, you can see the behavior in caja, press <kbd>ctrl</kbd><kbd>l</kbd> and you can see <kbd>tab</kbd> works as expected, it works for autocomplete and later it works changing widgets

**Pull request merged**